pkg/alertmanager/alertstore/store.go

```diff
@@ -40,6 +40,9 @@ type AlertStore interface {
 	// If configuration for the user doesn't exist, no error is reported.
 	DeleteAlertConfig(ctx context.Context, user string) error
 
+	// ListUsersWithState returns the list of users which have had state written.
+	ListUsersWithState(ctx context.Context) ([]string, error)
```

Comment:
[nit] Should we call it `ListUsersWithFullState` for consistency with other function names?